File: src/main/java/field/movement/fragment/ActionMoveFragment.java
package field.movement.fragment;

import field.object.life.FieldLife;
import util.packet.PacketReader;
import util.packet.PacketWriter;

public class ActionMoveFragment extends AbstractMovementFragment {

    byte moveAction;
    short elapse;

    public ActionMoveFragment(byte movePathAttribute, PacketReader packetReader) {
        super(movePathAttribute, packetReader);
    }

    @Override
    public void apply(FieldLife life) {
        life.setMoveAction(moveAction);
    }

    @Override
    public void decodeData(PacketReader packet) {
        moveAction = packet.readByte();
        elapse = packet.readShort();
    }

    @Override
    public void encodeData(PacketWriter packet) {
        packet.write(moveAction);
        packet.writeShort(elapse);
    }
}

File: src/main/java/field/movement/fragment/ActionMoveFragment.kt
package field.movement.fragment

import field.`object`.life.FieldLife
import util.packet.PacketReader
import util.packet.PacketWriter

open class ActionMoveFragment(movePathAttribute: Byte, packetReader: PacketReader) : AbstractMovementFragment(movePathAttribute, packetReader) {

    var moveAction: Byte = 0
    var elapse: Short = 0

    override fun apply(life: FieldLife) {
        life.moveAction = moveAction
    }

    override fun decodeData(packet: PacketReader) {
        moveAction = packet.readByte()
        elapse = packet.readShort()
    }

    override fun encodeData(packet: PacketWriter) {
        packet.write(moveAction.toInt())
        packet.writeShort(elapse)
    }
}